lib/kafka/connection.rb

```diff
@@ -88,10 +93,10 @@ def send_request(request)
 
         @correlation_id += 1
 
-        write_request(request, notification)
+        @operation.write_request(request, notification, @correlation_id)
 
         response_class = request.response_class
-        wait_for_response(response_class, notification) unless response_class.nil?
+        @operation.wait_for_response(response_class, notification, @correlation_id) unless response_class.nil?
```

Comment:
I can't make new instance of a Connection inside SaslConnectionWithTimeout. It call's `open` etc. So that was the reason I moved out just simple read/write and wrote own `send_request` in SaslConnection class. The other (dirty) way is to compose kafka message (SaslHandshakeRequest) out of the box and do not use this at all. But when protocol changes, it must be changes here as well, so it's really ugly and I didn't do that.

The reason why it in separate class - Kafka supports sasl plain auth too (and other sasl methods), which are not here as well and then encryption with SSL (after SASL auth). So I think it's better to have separate class, where I can do the SASL protocol negotiation (plain vs gssapi or others), do the auth and simply return the socket.